File: core/src/main/java/com/graqr/threshr/model/redsky/product/Video.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;
import jakarta.annotation.Nullable;

import java.util.List;

@Serdeable
public record Video(
        @JsonProperty("video_files") List<VideoFile> videoFiles,

        @Nullable @JsonProperty("video_captions") List<VideoCaption> videoCaptions,
        @Nullable @JsonProperty("video_length_seconds") String videoLengthSeconds,
        @Nullable @JsonProperty("video_poster_image") String videoPosterImage,
        @Nullable @JsonProperty("video_title") String videoTitle,
        @Nullable Boolean isListPageEligible) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/product/Video.kt
package com.graqr.threshr.circleStore.model.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable
import jakarta.annotation.Nullable

@Serdeable
data class Video(
    @param:JsonProperty("video_files") val videoFiles: List<VideoFile>,
    @param:Nullable @param:JsonProperty("video_captions") val videoCaptions: List<VideoCaption>,
    @param:Nullable @param:JsonProperty("video_length_seconds") val videoLengthSeconds: String,
    @param:Nullable @param:JsonProperty("video_poster_image") val videoPosterImage: String,
    @param:Nullable @param:JsonProperty("video_title") val videoTitle: String,
    @param:Nullable val isListPageEligible: Boolean
) {}
